src/cdk/commands/renderGraph.ts

```diff
@@ -0,0 +1,46 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import * as vscode from 'vscode'
+import { getLogger } from '../../shared/logger'
+import { localize } from '../../shared/utilities/vsCodeUtils'
+import { Commands } from '../../shared/vscode/commands'
+import { Window } from '../../shared/vscode/window'
+import { ConstructNode } from '../explorer/nodes/constructNode'
+import { showErrorWithLogs } from '../../shared/utilities/messages'
+import { AslVisualizationCDK } from './aslVisualizationCDK'
+
+/**
+ * Renders a state graph of the state machine represented by the given node
+ */
+export async function renderGraphCommand(
+    node: ConstructNode,
+    window = Window.vscode(),
+    commands = Commands.vscode()
+): Promise<AslVisualizationCDK | undefined> {
+    getLogger().debug('Render graph called for: %O', node)
+
+    const uniqueIdentifier = node.label
+    var cdkOutPath = node.id?.replace(`/tree.json/${node.tooltip}`, ``)!
+    var stackName = node.tooltip?.replace(`/${node.label}`, ``)!
+    getLogger().info(`Rendering graph: ${uniqueIdentifier}`)
+
+    try {
+        const templatePath = cdkOutPath + `/${stackName}.template.json`
+        let uri = vscode.Uri.file(templatePath);
+        const textDocument = await vscode.workspace.openTextDocument(uri)
+        const newVisualization = new AslVisualizationCDK(textDocument, templatePath, uniqueIdentifier)
```

Comment:
The visualization should be retrieved from the visualization manager instead of directly constructing the visualization here. We should just ask the visualization manager to visualize this particular state machine and it should either retrieve the existing one or create a new one for us.